spec/reek/smells/boolean_parameter_spec.rb

```diff
@@ -1,91 +1,103 @@
 require_relative '../../spec_helper'
 require_lib 'reek/smells/boolean_parameter'
-require_relative 'smell_detector_shared'
 
 RSpec.describe Reek::Smells::BooleanParameter do
-  context 'parameter defaulted with boolean' do
-    context 'in a method' do
-      it 'reports a parameter defaulted to true' do
-        src = 'def cc(arga = true); arga; end'
-        expect(src).to reek_of(:BooleanParameter, parameter: 'arga')
-      end
-
-      it 'reports a parameter defaulted to false' do
-        src = 'def cc(arga = false) end'
-        expect(src).to reek_of(:BooleanParameter, parameter: 'arga')
+  it 'reports the right values' do
+    src = <<-EOS
+      def m(a = true)
+        a
       end
+    EOS
+
+    expect(src).to reek_of(:BooleanParameter,
+                           lines: [1],
+                           context: 'm',
+                           message: "has boolean parameter 'a'",
+                           source: 'string',
+                           parameter: 'a')
+  end
 
-      it 'reports two parameters defaulted to booleans' do
-        src = 'def cc(nowt, arga = true, argb = false, &blk) end'
-        expect(src).to reek_of(:BooleanParameter, parameter: 'arga')
-        expect(src).to reek_of(:BooleanParameter, parameter: 'argb')
+  it 'does count all occurences' do
+    src = <<-EOS
+      def m1(a = true, b = true)
       end
 
-      it 'reports keyword parameters defaulted to booleans' do
-        src = 'def cc(arga: true, argb: false) end'
-        expect(src).to reek_of(:BooleanParameter, parameter: 'arga')
-        expect(src).to reek_of(:BooleanParameter, parameter: 'argb')
+      def m2(x = true, y = true)
       end
+    EOS
+
+    expect(src).to reek_of(described_class,
+                           lines:   [1],
+                           context: 'm1',
+                           parameter: 'a')
+    expect(src).to reek_of(described_class,
+                           lines:   [1],
+                           context: 'm1',
+                           parameter: 'b')
+    expect(src).to reek_of(described_class,
+                           lines:   [4],
+                           context: 'm2',
+                           parameter: 'x')
+    expect(src).to reek_of(described_class,
+                           lines:   [4],
+                           context: 'm2',
+                           parameter: 'y')
```

Comment:
I think using very short names for things makes it harder to check that these four expectations are actually correct.